kb/compiling-and-building/how-do-i-configure-mbedtls.md

```diff
@@ -24,7 +24,7 @@ The `config.py` script automatically finds the `mbedtls_config.h` file when it r
 
 You might want to keep the custom configuration file for your application outside the Mbed TLS source tree. You can do this by defining the macro `MBEDTLS_CONFIG_FILE` for the desired filename (including the quote or angular brackets) at compile time. For example, using **make**:
     ```
-     CFLAGS="-Ipath/to/config -DMBEDTLS_CONFIG_FILE='<my_config.h>'" make
+     CFLAGS="-I 'path/to/config' -DMBEDTLS_CONFIG_FILE='\"<my_config.h>\"'" make
```

Comment:
This change is wrong. It results in being equivalent to
```
#include "<my_config.h>"
```

The original code is correct as long as `path/to/config` doesn't contain special characters such as spaces.

Reply:
Little bit confused here. Today I used the following exact command to successfully compile an empty config file:
```CFLAGS="-I '/home/my_user/repos/scratch/mbedtls/configs' -DMBEDTLS_CONFIG_FILE='\"/home/my_user/repos/scratch/mbedtls/configs/empty-config.h\"'" make lib -j```

I am not sure I understand which bit you are saying is wrong?

Reply:
That's not the same command. The expansion of the macro `MBEDTLS_CONFIG_FILE` must be either `"filename.h"` or `<filename.h>`, i.e. the file name surrounded either by double quotes or angle brackets. The code in the documentation arranges for angle brackets, which is fine. The code in your latest comment arranges for double quotes, which is fine. The code in your commit puts both double quotes and angle brackets, which doesn't work.

The rest of the quoting are extra layers to get the desired characters through two levels of shell expansion: first the shell that runs make, then the shell that make runs.

Reply:
Thank you for the explanation. I get the bit about adding the extra layers to get the desired bits through the shell expansion (this is what I thought the original issue was). What I had failed to realise in the original docs is why the angle brackets were there. I thought they were being used in the placeholder manner which I often see in documentation so when I ran the command myself I removed them and just put in the name of my config file and then this didn't work because I didn't have the quotes or the angle brackets.

So, the original docs are fine, I misinterpreted them. Shall I just close this PR?

Reply:
I agree this is confusing. I'd be in favour of adding a note to point out that the angle brackets are included here (or switching to using `\"` for the include instead, as it's less likely to be misinterpreted).

Reply:
I haven't checked if I was actually the one who wrote this line, but I'd write it with `<>` because they're less of a pain to quote. With double quotes, you have to think about how you're going to quote them due to their effect on the shell.

> I'd be in favour of adding a note to point out that the angle brackets are included here

You mean, something like “including the quote or angular brackets”, in the sentence just above the code snippet?

Reply:
> You mean, something like “including the quote or angular brackets”, in the sentence just above the code snippet?

Yes exactly, something like "(note that the angle brackets `<>` are included)".